File: abstract-factory/go/src/vehicle/Bus.go
package vehicle

import (
  "fmt"
)

// 具体产品实现类
type Bus struct {
  Name string
}

func (b Bus) GetName() string {
  return b.Name
}

func (b *Bus) Run() {
  fmt.Println("Bus::Run() [Name=" + b.Name + "]")
}

File: abstract-factory/c/src/vehicle/bus.c
#include "../func.h"

/* Bus对象函数实现 */
void bus_run(Vehicle *vehicle)
{
  printf("\r\n bus_run() [name=%s]", vehicle->name);
}

/* 创建对象构造函数 */
Bus *bus_constructor(char *name)
{
  Bus *bus = (Bus *)malloc(sizeof(Bus));
  strncpy(bus->name, name, 50);
  bus->run = &bus_run;
  return bus;
}